pkg/snapshotUtils/snapshot.go

```diff
@@ -48,42 +49,74 @@ func checkPhasesAndSendResult(waitForPhases []backupdriverv1.SnapshotPhase, snap
 	}
 }
 
-func SnapshopRef(ctx context.Context, clientSet *v1.BackupdriverV1Client, objectToSnapshot core_v1.TypedLocalObjectReference, namespace string, repository BackupRepository, waitForPhases []backupdriverv1.SnapshotPhase, logger logrus.FieldLogger) (backupdriverv1.Snapshot, error) {
-	snapshotUUID, err := uuid.NewRandom()
+/*
+ * Create a Snapshot record in the specified namespace. The snapshot is created
+ * only if another one does not already exist.
+ */
+func SnapshopRef(ctx context.Context,
+	clientSet *v1.BackupdriverV1Client,
+	objectToSnapshot core_v1.TypedLocalObjectReference,
+	namespace string,
+	repository BackupRepository,
+	waitForPhases []backupdriverv1.SnapshotPhase,
+	logger logrus.FieldLogger) (backupdriverv1.Snapshot, error) {
 
+	// Check if the snapshot record had already been created
+	var snapshotCR *backupdriverv1.Snapshot
+
+	snapshotList, err := clientSet.Snapshots(namespace).List(metav1.ListOptions{})
 	if err != nil {
-		return backupdriverv1.Snapshot{}, errors.Wrapf(err, "Could not create snapshot UUID")
+		return backupdriverv1.Snapshot{}, errors.Errorf("Failed to list snapshots from the %v namespace", namespace)
 	}
-	snapshotCR := backupdriverv1.Snapshot{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Snapshot",
-			APIVersion: "backupdriver.io/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: snapshotUUID.String(),
-		},
-		Spec: backupdriverv1.SnapshotSpec{
-			TypedLocalObjectReference: objectToSnapshot,
-			BackupRepository:          repository.backupRepository,
-			SnapshotCancel:            false,
-		},
+	logger.Infof("Found %d snapshots in namespace %s", len(snapshotList.Items), namespace)
+	for _, snapItem := range snapshotList.Items {
```

Comment:
it would introduce extra performance concerns. And taking snapshot would be slower as the number of snapshot in cluster grows.